src/frontend/apidocs/ApiDocsFrame.js

```diff
@@ -2,11 +2,60 @@ import React, { Component } from "react";
 import PropTypes from "prop-types";
 import SwaggerUi from "swagger-ui";
 
+const ShowChangesPlugin = function (system) {
+  return {
+    wrapComponents: {
+      info: (Original, system) => class WrappedNumberDisplay extends Component {
+        constructor(props) {
+          super(props);
+
+          this.state = { isOpened: false };
+          this.changes = props.info.get('x-changes');
+        }
+
+        render() {
+          const Collapse = system.getComponent("Collapse");
+
+          return (
+            <div>
+              <Original {...this.props} />
+              <h2>API Changelog <button className="swagger-ui btn" onClick={() => this.setState({ isOpened: !this.state.isOpened })}>{this.state.isOpened ? "Hide" : "Show"}</button></h2>
+              <Collapse isOpened={this.state.isOpened}>
+                <ChangesComponent changes={this.changes} />
+              </Collapse>
+            </div>
+          );
+        }
+      }
+    }
+  }
+}
+
+class ChangesComponent extends Component {
+  render() {
+    return this.props.changes.map((changes, version) => {
```

Comment:
This is broken because `this.props.changes` is an ImmutableJS OrderedMap, so `.map` doesn't really behave as expected.

You'll need to do this to convert the OrderedMap into a normal JS array:
```
    return this.props.changes
      .entrySeq()
      .toArray()
      .map((item) => {
        const version = item[0];
        const changes = item[1].toArray();
        return (
          <div key={version}>
            <h3>{version}</h3>
            <ul>
              {changes.map((c, i) => (
                <li key={i}>{c}</li>
              ))}
            </ul>
          </div>
        );
      });
```

Also when you render a list with `.map`, each child needs a `key` https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key

Reply:
edit: actually both of the `.toArray()` are unnecessary